cns/deviceplugin/pluginmanager_test.go

```diff
@@ -0,0 +1,283 @@
+			return d.DialContext(ctx, "unix", addr)
+		}))
+	if err != nil {
+		t.Fatalf("error connecting to fake kubelet: %v", err)
+	}
+	defer conn.Close()
+
+	client := v1beta1.NewDevicePluginClient(conn)
+	lwClient, err := client.ListAndWatch(context.Background(), &v1beta1.Empty{})
+	if err != nil {
+		t.Fatalf("error from listAndWatch: %v", err)
+	}
+
+	resp, err := lwClient.Recv()
+	if err != nil {
+		t.Fatalf("error from listAndWatch Recv: %v", err)
+	}
+
+	return len(resp.Devices)
+}
+
+//nolint:staticcheck // TODO: Move to grpc.NewClient method
+func getAllocateResponse(t *testing.T, pluginAddress string, req *v1beta1.AllocateRequest) *v1beta1.AllocateResponse {
+	conn, err := grpc.Dial(pluginAddress, grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			d := &net.Dialer{}
+			return d.DialContext(ctx, "unix", addr)
+		}))
+	if err != nil {
+		t.Fatalf("error connecting to fake kubelet: %v", err)
+	}
+	defer conn.Close()
+
+	client := v1beta1.NewDevicePluginClient(conn)
+	resp, err := client.Allocate(context.Background(), req)
+	if err != nil {
+		t.Fatalf("error from Allocate: %v", err)
+	}
+	return resp
+}
+
+//nolint:staticcheck // TODO: Move to grpc.NewClient method
+func getDevicePluginOptionsResponse(t *testing.T, pluginAddress string) (*v1beta1.DevicePluginOptions, error) {
+	conn, err := grpc.Dial(pluginAddress, grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			d := &net.Dialer{}
+			return d.DialContext(ctx, "unix", addr)
+		}))
+	if err != nil {
+		t.Fatalf("error connecting to fake kubelet: %v", err)
+	}
+	defer conn.Close()
+
+	client := v1beta1.NewDevicePluginClient(conn)
+	resp, err := client.GetDevicePluginOptions(context.Background(), &v1beta1.Empty{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error calling GetDevicePluginOptions")
+	}
+	return resp, nil
+}
+
+//nolint:staticcheck // TODO: Move to grpc.NewClient method
+func getPreferredAllocationResponse(t *testing.T, pluginAddress string) (*v1beta1.PreferredAllocationResponse, error) {
+	conn, err := grpc.Dial(pluginAddress, grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			d := &net.Dialer{}
+			return d.DialContext(ctx, "unix", addr)
+		}))
+	if err != nil {
+		t.Fatalf("error connecting to fake kubelet: %v", err)
+	}
+	defer conn.Close()
+
+	client := v1beta1.NewDevicePluginClient(conn)
+	resp, err := client.GetPreferredAllocation(context.Background(), &v1beta1.PreferredAllocationRequest{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error calling GetPreferredAllocation")
+	}
+	return resp, nil
+}
+
+//nolint:staticcheck // TODO: Move to grpc.NewClient method
+func getPreStartContainerResponse(t *testing.T, pluginAddress string) (*v1beta1.PreStartContainerResponse, error) {
```

Comment:
Usually it gets weird if you're calling `t.Fatal` and returning errors as well. I would just return errors and leave it up to the caller to `t.Fatal`.

Reply:
updated